ostd/src/cpu/id.rs

```diff
@@ -0,0 +1,237 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! CPU identification numbers.
+
+pub use current::PinCurrentCpu;
+
+/// The ID of a CPU in the system.
+///
+/// If converting from/to an integer, the integer must start from 0 and be less
+/// than the number of CPUs.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct CpuId(u32);
+
+impl CpuId {
+    /// Creates a new instance.
+    ///
+    /// # Panics
+    ///
+    /// The given number must be within the range `[0..ostd::cpu::num_cpus()]`;
+    /// otherwise, the method will panic.
+    pub fn new(raw_id: u32) -> Self {
+        assert!(raw_id < num_cpus() as u32);
+        // SAFETY: the above assertion ensures that the raw number is valid.
+        unsafe { Self::new_unchecked(raw_id) }
+    }
+
+    /// Creates a new instance.
+    ///
+    /// # Safety
+    ///
+    /// The given number must be within the range `[0..ostd::cpu::num_cpus()]`.
+    pub(super) unsafe fn new_unchecked(raw_id: u32) -> Self {
+        // Is this a good idea to mark this function as const?
+        //
+        // Not really. This is because
+        // the number of CPU cores can only be determined at the run time.
+        // As a result, it is impossible to use `CpuId::new_unchecked`
+        // both statically and safely.
+
+        Self(raw_id)
+    }
+
+    /// Returns the CPU ID of the bootstrap processor (BSP).
+    ///
+    /// The number for the BSP is always zero.
+    pub const fn bsp() -> Self {
+        // SAFETY:
+        // (1) there is at least one CPU;
+        // (2) BSP's `CURRENT_CPU` is assigned a value of 0.
```

Comment:
```suggestion
        // SAFETY: The number of CPUs are always at least one.
```
- I'm not aware of any existing practices that use `(1)` in the safety comments. We use `1. xxx` and `2. yyy` in many other places, but not `(1)` and `(2)`:
```
> grep -R "// (1) " ostd/src
ostd/src/arch/x86/mod.rs:/// (1) The `cvm_guest` feature is enabled at compile time.
```
- I'm also not aware of any existing practices that use lowercase letters for the first letter of safety comments.
- The second point is unrelated. It's the correctness guarantee of this method, but it's not the safety requirement of `new_unchecked`.